### PR TITLE
Initial draft of Chapter 3

### PR DESCRIPTION
**Pull request by zacharyburnett**

### CODE REVIEW

File: notebooks/04-Creating_ASDF_Files.ipynb [hunk 1]
```diff
@@ -0,0 +1,753 @@
+{
```
Comment (perrygreenfield) [inline notebooks/04-Creating_ASDF_Files.ipynb:1]: <p>Should include boolean type as part of the type list</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/asdf-format/scipy2022tutorial/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (perrygreenfield) [inline notebooks/04-Creating_ASDF_Files.ipynb:1]: <p>Should we standardize what the variable name is for the <code>AsdfFile</code> object? For example I use <code>af</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/asdf-format/scipy2022tutorial/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (perrygreenfield) [inline notebooks/04-Creating_ASDF_Files.ipynb:1]: <p>Do we need the mode keyword? That's the default value.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/asdf-format/scipy2022tutorial/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (perrygreenfield) [inline notebooks/04-Creating_ASDF_Files.ipynb:1]: <p>I don't think the mode keyword is needed for read mode.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/asdf-format/scipy2022tutorial/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (perrygreenfield) [inline notebooks/04-Creating_ASDF_Files.ipynb:1]: <p>Maybe mention that updates may be expensive if they require rewriting the whole file (refer to padding and other mitigation strategies elsewhere, or that these can be avoided).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/asdf-format/scipy2022tutorial/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (WilliamJamieson) [reply notebooks/04-Creating_ASDF_Files.ipynb:1]: <p><code>mode</code> must be used if you want to specify the open method. Well the issue is more complicated, there is another option before <code>mode</code> in the interface, so naively specifying without the keyword is dangerous.</p>

**Comment by zacharyburnett** [comment]
@WilliamJamieson if this has addressed the comments do you think it is ready to merge?

**Comment by WilliamJamieson** [comment]
> @WilliamJamieson if this has addressed the comments do you think it is ready to merge?

Lets wait for @perrygreenfield or @nden to formally approve the PR.

**Review by perrygreenfield — APPROVED** [review]
Just a minor note about the explanation of the mode keyword. I think we were talking past each other. As used, it isn't a method, it's a module function. I agree that if you need to specify mode, the keyword form is recommended. But if it is read only, as far as I know, it doesn't need to be supplied and it can be omitted.

**Pull request merged**